utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -457,130 +527,153 @@ pub enum State {
-		#[arg(
-			short,
-			long,
-			value_parser = parse::hash,
-		)]
-		at: Option<String>,
-
-		/// An optional state snapshot file to WRITE to. Not written if set to `None`.
-		#[arg(short, long)]
-		snapshot_path: Option<PathBuf>,
-
-		/// A pallet to scrape. Can be provided multiple times. If empty, entire chain state will
-		/// be scraped.
-		#[arg(short, long, num_args = 1..)]
-		pallet: Vec<String>,
-
-		/// Fetch the child-keys as well.
-		///
-		/// Default is `false`, if specific `--pallets` are specified, `true` otherwise. In other
-		/// words, if you scrape the whole state the child tree data is included out of the box.
-		/// Otherwise, it must be enabled explicitly using this flag.
-		#[arg(long)]
-		child_tree: bool,
-	},
+	Live(LiveState),
 }
 
 impl State {
-	/// Create the [`remote_externalities::Builder`] from self.
-	pub(crate) fn builder<Block: BlockT + DeserializeOwned>(&self) -> sc_cli::Result<Builder<Block>>
+	/// Create the [`sp_io::TestExternalities`] using [`remote-externalities`] from self.
+	///
+	/// This will override the code as it sees fit based on [`SharedParams::Runtime`]. It will also
+	/// check the spec-version and name.
+	pub(crate) async fn into_ext<Block: BlockT + DeserializeOwned, HostFns: HostFunctions>(
+		&self,
+		shared: &SharedParams,
+		config: &Configuration,
+		executor: &WasmExecutor<HostFns>,
+	) -> sc_cli::Result<RemoteExternalities<Block>>
 	where
 		Block::Hash: FromStr,
+		Block::Header: DeserializeOwned,
+		Block::Hash: DeserializeOwned,
 		<Block::Hash as FromStr>::Err: Debug,
 	{
-		Ok(match self {
+		let builder = match self {
 			State::Snap { snapshot_path } =>
 				Builder::<Block>::new().mode(Mode::Offline(OfflineConfig {
 					state_snapshot: SnapshotConfig::new(snapshot_path),
 				})),
-			State::Live { snapshot_path, pallet, uri, at, child_tree } => {
+			State::Live(LiveState { pallet, uri, at, child_tree, threads }) => {
 				let at = match at {
 					Some(at_str) => Some(hash_of::<Block>(at_str)?),
 					None => None,
 				};
-				let mut builder = Builder::<Block>::new()
-					.mode(Mode::Online(OnlineConfig {
-						transport: uri.to_owned().into(),
-						state_snapshot: snapshot_path.as_ref().map(SnapshotConfig::new),
-						pallets: pallet.clone(),
-						scrape_children: true,
-						at,
-					}))
-					.inject_hashed_key(
-						&[twox_128(b"System"), twox_128(b"LastRuntimeUpgrade")].concat(),
-					);
-				if *child_tree {
-					builder = builder.inject_default_child_tree_prefix();
-				}
-				builder
+				Builder::<Block>::new().mode(Mode::Online(OnlineConfig {
+					at,
+					transport: uri.to_owned().into(),
+					state_snapshot: None,
+					pallets: pallet.clone(),
+					child_trie: *child_tree,
+					hashed_keys: vec![
+						// we always download the code, but we almost always won't use it, based on
+						// `Runtime`.
+						well_known_keys::CODE.to_vec(),
+						// we will always download this key, since it helps detect if we should do
```

Comment:
question: where do we actually use it?

Reply:
[It happens inside the runti](https://github.com/paritytech/substrate/blob/dd8a34775aab9b362a5cc7552d5603c2fe812f82/frame/executive/src/lib.rs#L379)